### DISCUSSION TITLE
Suggestion on MCMC.

### DISCUSSION THREAD

**Discussion by mamun-nahid** [AstroBridge/BDLensing#197] [Q&A]
I am facing some issues with MCMC. It's not reproducing the lower arc at all and there are residuals on the upper arc too. 
Here's the notebook [link](https://colab.research.google.com/drive/1FJzT5hbtVyG4-Q1SotpOBzqZgOw0Pm5s?usp=sharing).

![Untitled1](https://github.com/ajshajib/BDLensing/assets/61522619/ec9438e4-d3f7-49ab-90df-c380d3381f74)

I have tried masking out the lower arc as it may not be part of the source light. That doesn't seem to work either.  Notebook [link](https://colab.research.google.com/drive/113HUshcmREMZZ6s9U4NSGhLYntbzEhTj?usp=sharing).

![Untitled](https://github.com/ajshajib/BDLensing/assets/61522619/11ddf8b7-c31b-46a5-8c7c-5bead4ba9bb5)

Any suggestion on how it can be improved?

**Comment by ajshajib** [comment]
Have you run PSO before the MCMC? The MCMC needs to start from the lens model that PSO gives you.

**Reply by mamun-nahid** [reply]
Yes. I have run PSO before MCMC. but I have loaded the saved PSO joblib file before running MCMC. Is it causing any problems?
